src/components/picker/types.tsx

```diff
@@ -147,49 +130,112 @@ export type PickerBaseProps = Omit<NewTextFieldProps, 'value' | 'onChange'> & {
   /**
-   * Pass props to the list component that wraps the picker options (allows to control FlatList behavior)
+   * Adds blur effect to picker modal (iOS only)
    */
-  listProps?: Partial<FlatListProps<any>>;
+  enableModalBlur?: boolean;
+};
+
+type PickerListProps = {
   /**
-   * Pass props to the picker modal
+   * Render a custom header for Picker's Overlay
    */
-  pickerModalProps?: object;
+  renderCustomOverlayHeader?: (callbacks: {onDone?: () => void; onCancel?: () => void}) => React.ReactElement;
   /**
-   * Custom container style
-   */
-  containerStyle?: StyleProp<ViewStyle>;
-  /**
-   * @deprecated
-   * Callback for modal onShow event
+   * Pass props to the list component that wraps the picker options (allows to control FlatList behavior)
    */
-  onShow?: () => void;
+  listProps?: Partial<FlatListProps<any>>;
   /**
    * Add safe area in the Picker modal view
    */
   useSafeArea?: boolean;
-  /**
-   * Data source for Picker
-   */
-  items?: PickerItemProps[];
-  /**
-   * Component test id
-   */
-  testID?: string;
-  /**
-   * @deprecated
-   */
-  children?: ReactNode | undefined;
 };
 
+export type PickerBaseProps = Omit<NewTextFieldProps, 'value' | 'onChange'> &
+  PickerSearchProps &
+  PickerPropsDeprecation &
+  PickerExpandableOverlayProps &
+  PickerListProps & {
+    /* ...TextField.propTypes, */
+    /**
+     * Use dialog instead of modal picker
+     */
+    useDialog?: boolean;
+    /**
+     * Pass for different field type UI (form, filter or settings)
+     */
+    fieldType?: PickerFieldTypes | `${PickerFieldTypes}`;
+    /**
+     * Picker current value in the shape of {value: ..., label: ...}, for custom shape use 'getItemValue' prop
+     */
+    value?: PickerValue;
+    /**
+     * Callback for when picker value change
+     */
+    onChange?: (value: PickerValue) => void;
+    /**
+     * SINGLE mode or MULTI mode
+     */
+    mode?: PickerModes | `${PickerModes}`;
+    /**
+     * Limit the number of selected items
+     */
+    selectionLimit?: number;
+    /**
+     * Render custom picker input
+     * Example:
+     * renderInput = (selectedItem) => {...}
+     */
+    renderInput?: RenderPicker;
+    /**
+     * Render custom picker item
+     */
+    renderItem?: (
+      value: PickerValue,
+      itemProps: PickerItemProps & {isSelected: boolean; isItemDisabled: boolean},
+      label?: string
+    ) => React.ReactElement;
+    /**
+     * Add onPress callback for when pressing the picker
+     */
+    onPress?: () => void;
+    /**
+     * A function that returns the label to show for the selected Picker value
+     */
+    getLabel?: (value: PickerValue) => string;
+    /**
+     * The picker modal top bar props
+     */
+    topBarProps?: ModalTopBarProps;
+    /**
+     * Use wheel picker instead of a list picker
+     */
+    useWheelPicker?: boolean;
+    /**
+     * Custom picker input container style
+     */
+    inputContainerStyle?: StyleProp<ViewStyle>;
```

Comment:
containerStyle is better - I wouldn't change the name here

Reply:
renamed back to containerStyle